File: Day-1/Day1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;

int main() {
    string input;
    // Read input from file
    ifstream file("Day1.txt");

    if (file.is_open()) {
        
        int total = 0;
        string temp = "";

        while (getline(file, input)) {
            
            string firstDigit = "";
            string lastDigit = "";

            for(int i = 0; i < input.length(); i++) {

                if(isdigit(input[i])) {
                    if(firstDigit.empty()) {
                        cout << firstDigit;
                        firstDigit = input[i];
                    }
                    lastDigit = input[i];
                }
                
            }
            cout << "First digit is " << firstDigit << " and last digit is " << lastDigit << "\n";
            total += stoi(firstDigit+lastDigit);

        }

    file.close();
    std::cout << total;

    }
    return 0;

}

//60632

File: Day-1/Day1.py
def sum_calibration_values_from_file(file_path):
    total_sum = 0

    with open(file_path, 'r') as file:
        for line in file:
            # Filter out non-digit characters
            digits = [char for char in line if char.isdigit()]

            # Check if we have at least one digit in the line
            if digits:
                # Combine the first and last digits to form a two-digit number
                value = int(digits[0] + digits[-1])

                # Add the value to the total sum
                total_sum += value

    return total_sum

# Replace 'Day1.txt' with the actual path to your file
file_path = 'Day1.txt'

# Calculate the sum of calibration values from the file
result = sum_calibration_values_from_file(file_path)

# Print the result
print("Sum of calibration values:", result)

